pkg/controller/cluster/main-controller.go

```diff
@@ -957,17 +965,26 @@ func (c *Controller) syncHandler(key string) error {
 		if err != nil {
 			_ = c.removeArtifacts()
 
+			err = fmt.Errorf("Tenant '%s' MinIO update failed with %w", name, err)
+			if _, terr := c.updateTenantStatus(ctx, mi, err.Error(), totalReplicas); terr != nil {
+				return terr
+			}
+
 			// Update failed, nothing needs to be changed in the container
-			return fmt.Errorf("Tenant '%s' MinIO update failed with %w", name, err)
+			return err
 		}
 
 		if us.CurrentVersion != us.UpdatedVersion {
 			klog.Infof("Tenant '%s' MinIO updated successfully from: %s, to: %s successfully",
 				name, us.CurrentVersion, us.UpdatedVersion)
 		} else {
-			klog.Infof("Tenant '%s' MinIO is already running the most recent version of %s",
+			msg := fmt.Sprintf("Tenant '%s' MinIO is already running the most recent version of %s",
```

Comment:
but if we do this, we are never going back to ready state are we?

Reply:
@dvaldivia yes we are never ready if update failed.

Reply:
makes sense, but it's not an static status state, I think we can improve the status in the future to have status + message so we can provide details on why we reached a certain state